### PR TITLE
introduce new source detail option eventhub.accessKeySecretName

### PR DESCRIPTION
**Pull request by msdotnetclr**
Introduce a new source_details option "eventhub.accessKeySecretName" to store the name of the secret holding the actual Event Hub access key (Shared Access Policy) value, keep "eventhub.accessKeyName" to store the name of the access key (Shared Access Policy). 

This allows the code to use "eventhub.accessKeySecretName" to get the actual access key value, and use "eventhub.accessKeyName" to get the access key name, thereby construct eh_shared_key_value properly, regardless if "eventhub.accessKeySecretName" and "eventhub.accessKeyName" are the same.

### CODE REVIEW

**Comment by msdotnetclr** [comment]
Link this to issue #13

**Comment by msdotnetclr** [comment]
Pushed a few more commits to address the following:
- Backward compatibility (so existing event hubs pipelines that are already working won't break when the changes are deployed)
- Unit test and integration test
- Documentation
Please review.

**Comment by rportilla-databricks** [comment]
3 updates that we will need to make according to the repo owner: 

1 Remove oryx-build-commands.txt file
2) Update the changelog (suggested update)

## [v.0.0.4] - 2023-10-09
### Added 
-  Functionality to introduce an new option for event hub configuration. Namely a source_details option 'eventhub.accessKeySecretName' to properly construct the eh_shared_key_value properly. Without this option, there were errors while connecting to the event hub service (linked to issue-13 - https://github.com/databrickslabs/dlt-meta/issues/13)

3) Increment version to 0.0.4 inside setup.py

**Comment by msdotnetclr** [comment]
Thanks @rportilla-databricks. The suggested changes have been committed.

**Review by ravi-databricks — APPROVED** [review]
LGTM.

**Review by ganeshchand — APPROVED** [review]
Ran the integration test and wen through the changes. LGTM.

File: oryx-build-commands.txt
[the remark this answers is not in the document]
Comment (msdotnetclr) [reply oryx-build-commands.txt:1]: @ravi-databricks for some reason I was not able to see any of the comments above until last night. Curious if it was me being ignorant or something else. Thanks for all the feedback!

**Pull request merged**